repos/keg-core/core/scripts/js/getKegEnvs.js

```diff
@@ -0,0 +1,88 @@
+const { get, isStr } = require('@keg-hub/jsutils')
+const getAppConfig = require('@keg-hub/tap-resolver/src/resolvers/getAppConfig')
+const fs = require('fs')
+const path = require('path')
+
+/**
+ * @param {string} tapPath - path to the current tap's root
+ * @param {string} corePath - path to keg-core's root
+ * @return {Object} - object containing tap and core's app configs and package json objects
+ *  - used to resolve replacement values for envs of form "<tap|core>.<package|config>.<some_property>"
+ */
+const buildReplaceContext = (tapPath, corePath) => {
+  const buildContext = rootPath => ({
+    package: rootPath
+      ? require(path.join(rootPath, 'package.json') )
+      : {},
+    config: rootPath
+      ? getAppConfig(rootPath, false, false)
+      : {}
+  })
+
+  return {
+    core: buildContext(corePath),
+    tap: buildContext(tapPath),
+  }
+}
+
+/**
+ * 
+ * @param {Object} replaceContext - context object produced by `buildReplaceContext`
+ * @return {Object} - object of env entries pulled from tap config and core config, 
+ * dynamically replacing any values of form `<tap|core>.<package|config>.<some_property>`
+ */
+const buildEnvs = replaceContext => {
+  const tapEnvs = get(replaceContext, 'tap.config.keg.envs')
+  const coreEnvs = get(replaceContext, 'core.config.keg.envs')
+
+  const envs = {
+    ...coreEnvs,
+    ...tapEnvs
+  }
+
+  const getFallback = value => 
+    isStr(value) && (value.startsWith('tap.') || value.startsWith('core.'))
```

Comment:
I'm not sure I understand the point of this?
Why not just use `value` as the fallback
If it doesn't exist, then it will be `undefined` anyways

Reply:
In the case of the dynamic values, like `tap.package.version`, if there isn't a value for it, then returning `value` is just returning `tap.package.verson`, which is not undefined. This just makes sure that it does.

Reply:
Ahh got it. Thanks

Reply:
I'll update the variable names here to try to be clearer.